pkg/ccip/bindings/common/common.go

```diff
@@ -28,6 +29,26 @@ const (
 	DispatchNotFromMerkleRoot
 )
 
+const (
+	DestChainConfigGetter = "destChainConfig"
+	VersionGetter         = "typeAndVersion"
+	DestChainsGetter      = "destChainSelectors"
+	SrcChainConfigGetter  = "sourceChainConfig"
+)
+
+type FetchOptions struct {
+	SrcChainSelector  uint64
+	DestChainSelector uint64
+}
+
+// ConfigFetcher is an interface for fetching and parsing contract configurations.
+type ConfigFetcher interface {
+	// FetchResult fetches the configuration from the contract at the specified block and address.
+	FetchResult(ctx context.Context, client ton.APIClientWrapped, block *ton.BlockIDExt, contractAddr *address.Address, opts *FetchOptions) error
```

Comment:
This interface needs some polish.

Can fetch any result from any contract with these options `FetchOptions`?

Think this needs to be generalized with options T, so any getter can be provided for and called using this kind of API. Also, the implementations of this `FetchResult` interface are very similar to each other and should be easy to deduplicate.